prometheus-proc/src/Prometheus/Metric/Proc.hs

```diff
@@ -0,0 +1,175 @@
+{-# language ForeignFunctionInterface #-}
```

Comment:
Uh oh.

Reply:
Could avoid this by using [System.CPUTime cpuTimePrecision](https://github.com/ghc/packages-base/blob/52c0b09036c36f1ed928663abb2f295fd36a88bb/System/CPUTime.hsc#L153-L163), which encapsulates the FFI necessary to access `clk_tck`?

Reply:
Happy to change to this, but it's not immediately obvious how I would use it. That big modulus is just a kind of wrap-around, right? That is, I can just drop in `cpuTimePrecision`?